inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/util/SpecialElementMatchers.java

```diff
@@ -0,0 +1,135 @@
+package rocks.inspectit.oce.core.instrumentation.config.util;
+
+import net.bytebuddy.description.ModifierReviewable;
+import net.bytebuddy.description.NamedElement;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.NameMatcher;
+import net.bytebuddy.matcher.StringMatcher;
+import org.apache.commons.lang3.StringUtils;
+import rocks.inspectit.oce.core.config.model.instrumentation.scope.NameMatcherSettings;
+import rocks.inspectit.oce.core.config.model.instrumentation.scope.TypeScope;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+import static rocks.inspectit.oce.core.config.model.instrumentation.scope.MethodMatcherSettings.AccessModifier;
+
+/**
+ * This class provides advanced and custom ElementMatchers.
+ */
+public class SpecialElementMatchers {
+
+    private SpecialElementMatchers() {
+    }
+
+    /**
+     * Creates an {@link ElementMatcher} matching items with the given name settings.
+     */
+    public static <T extends NamedElement> ElementMatcher.Junction<T> nameIs(NameMatcherSettings settings) {
+        if (settings == null) {
+            return null;
+        }
+
+        String namePattern = settings.getName();
+
+        if (StringUtils.isNotEmpty(namePattern)) {
+            StringMatcher.Mode matcherMode = settings.getMatcherMode();
+            return new NameMatcher<>(new StringMatcher(namePattern, matcherMode));
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Creating an {@link ElementMatcher} matching methods which takes arguments equals to the given array. The array consists
+     * of String representing full qualified class names.
+     */
+    public static <T extends MethodDescription> ElementMatcher.Junction<T> argumentsAre(String[] arguments) {

```

Comment:
I think again that it is better to use a List instead of an array as argument here